### PR TITLE
On demand JS

### PR DESCRIPTION
**Pull request by trasher**
A try to load some JS "on demand".

On computers list, page length was 1589 Ko and is now 907 ko.

### CODE REVIEW

**Comment by ddurieux** [comment]
> On computers list, page length was 1589 Ko and is now 907 ko.
In debug? because is normal it's minified js files charged ;)

**Comment by trasher** [comment]
Not in debug, no ;)

**Comment by orthagh** [comment]
on single ticket view, js error:
```
Uncaught ReferenceError: filter_timeline is not defined(…)
```

File: inc/entity.class.php [hunk 1]
```diff
@@ -1064,7 +1064,9 @@ static function showSelector($target, $myname) {
       echo "</form>";
 
       echo "<script type='text/javascript'>";
-      echo Html::jsGetElementbyID("tree_projectcategory$rand")."
+      echo "   $(function() {
+                  $.getScript('{$CFG_GLPI["root_doc"]}/lib/jqueryplugins/jstree/jquery.jstree.min.js', function(data, textStatus, jqxhr) {
+                     $('#tree_projectcategory$rand')
          // call `.jstree` with the options object
```
Comment (trasher) [inline inc/entity.class.php]: I do not reindent; just add the required stuff around :D

File: inc/ajax.class.php [hunk 2]
```diff
@@ -209,9 +210,10 @@ static function createIframeModalWindow($domid, $url, $options=array() ) {
       $url .= (strstr($url, '?') ?'&' :  '?').'_in_modal=1';
 
       $out  = "<div id=\"$domid\">";
-      $out .= "<iframe id='Iframe$domid' class='iframe'></iframe></div>";
+      $out .= "<iframe id='Iframe$domid' class='iframe hidden'></iframe></div>";
 
-        $out .= "<script type='text/javascript'>
+      $out .= "<script type='text/javascript'>
+         $(function(){
```
Comment (orthagh) [inline inc/ajax.class.php]: missing space before brace

File: inc/html.class.php [hunk 3]
```diff
@@ -1729,7 +1712,7 @@ static function header($title, $url='', $sector="none", $item="none", $option=""
 
       // Back to top
       Html::scriptStart();
-      echo "$(document).ready(function(){
+      echo "$(function(){
```
Comment (orthagh) [inline inc/html.class.php]: missing space before brace

**Pull request merged**